cpp/src/arrow/array/builder_primitive.h

```diff
@@ -32,6 +32,14 @@ class ARROW_EXPORT NullBuilder : public ArrayBuilder {
   explicit NullBuilder(MemoryPool* pool ARROW_MEMORY_POOL_DEFAULT)
       : ArrayBuilder(null(), pool) {}
 
+  /// \brief Append the specified number of null elements
+  Status AppendNulls(int64_t length) {
+    null_count_ += length;
+    length_ += length;
```

Comment:
I think it is better to check overflow here, but I didn't because other builders don't do that.

Reply:
I found the overflow check is performed in `CheckCapacity()` function.